src/hooks/useOriginalReportID.ts

```diff
@@ -0,0 +1,29 @@
+import {getOneTransactionThreadReportID} from '@libs/ReportActionsUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {OnyxInputOrEntry, ReportAction} from '@src/types/onyx';
+import useOnyx from './useOnyx';
+
+function useOriginalReportID(reportID: string | undefined, reportAction: OnyxInputOrEntry<ReportAction>): string | undefined {
+    const [reportActions] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID ?? CONST.DEFAULT_NUMBER_ID}`, {canBeMissing: true});
+    const [report] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${reportID ?? CONST.DEFAULT_NUMBER_ID}`, {canBeMissing: true});
+    const [chatReport] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${report?.chatReportID}`, {canBeMissing: true});
+
+    if (!reportID) {
+        return undefined;
+    }
+    const currentReportAction = reportAction?.reportActionID ? reportActions?.[reportAction.reportActionID] : undefined;
+
+    if (Object.keys(currentReportAction ?? {}).length === 0) {
+        const isThreadReportParentAction = reportAction?.childReportID?.toString() === reportID;
+        if (isThreadReportParentAction) {
+            return report?.parentReportID;
+        }
+        const transactionThreadReportID = getOneTransactionThreadReportID(report, chatReport, reportActions ?? ([] as ReportAction[]));
+        const retorno = transactionThreadReportID ?? reportID;
+        return retorno;
+    }
+    return reportID;
+}
```

Comment:
Added this hook analog to the function [ReportUtils::getOriginalReportID](https://github.com/Expensify/App/blob/a5df0f340db9984d92412f33fde0a551ae42f72b/src/libs/ReportUtils.ts#L9091-L9106) which was reading data from `Onyx.connect` and causing this: https://github.com/Expensify/App/pull/68692#issuecomment-3206108322